### PR TITLE
shell: Hide remote dashboards

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Review by martinpitt — COMMENTED** [review]
I agree to the final result, but this is quite a hack. I wonder if there would be a more elegant way of declaring this in the manifest somehow?

**Comment by martinpitt** [comment]
Also, can you please add an integration test?

**Comment by mvollmer** [comment]
> I agree to the final result, but this is quite a hack. I wonder if there would be a more elegant way of declaring this in the manifest somehow?

Then we need a new release everywhere immediately.

We should retire the whole "dashboard" section eventually and move applications into "application".

**Comment by mvollmer** [comment]
> We should retire the whole "dashboard" section eventually and move applications into "application".

Hmm, not retire, but reserve it for the one and only true Cockpit Dashboard.

**Comment by marusak** [comment]
Isn't that going to be super confusing if you install `cockpit-dashboard` on your remote machine but don't see it there?

**Comment by mvollmer** [comment]
> Isn't that going to be super confusing if you install cockpit-dashboard on your remote machine but don't see it there?

You will see it if you log in directly to that machine.

The main problem is the dashboard was meant to be global, like the shell itself.  We don't have a shell for each remote machine, we only have a single global one. Applications like the image composer or fleet commander run on a specific machine. By accidents of history, the dashboard is now listed with the applications, and it has never made sense there from a navigational point of view. It could be a link the host switcher, maybe.

What's worse is that if you use a dashboard from a remote machine, it will load the code installed on that machine but access the bridge running on the local machine, which can be from a different version of Cockpit. It will read and write /etc/cockpit/machines.d of the local machine, for example. I don't think we want to support that, especially since a remote dashboard doesn't do anything useful beyond what the local dashboard already does.

There is no point to run the current cockpit-dashboard from a remote machine, but it might be problematic. One might argue that code loaded from a remote machine shouldn't even get the kind of access that Cockpit gives it currently.  (Namely, opening channels to any host that the user has logged in to.)

**Comment by marusak** [comment]
> The main problem is the dashboard ...

Right, all of that makes perfect sense. Here is an idea - what about obsoleting the whole dashboard? We have host switcher which covers all that functionality. There is that graph which we may want to keep? But then I suggest picking it into separate shell page and just putting `Graphs` button into the host switcher. That would solve all the troubles with dashboard.

**Comment by mvollmer** [comment]
> Here is an idea - what about obsoleting the whole dashboard?

Sure, by all means.  But we can also have this little hack to keep out us out of trouble while we work on that, no?

**Comment by martinpitt** [comment]
From my POV I'm ok with the hack, as we have a longer-term agreement now. I'd just still like to see a little integration test. We already have some that run a remote session, should suffice to check that the Dashboard menu entry doesn't exist.

File: test/verify/check-host-switching [hunk 1]
```diff
@@ -258,6 +261,9 @@ class TestHostSwitching(MachineCase, HostSwitcherHelpers):
         b.click("#nav-hosts .nav-item a[href='/@10.111.113.3']")
         b.wait_js_cond('window.location.pathname.indexOf("/@10.111.113.3") === 0')
 
+        # remote hosts should not have the dashboard
+        b.wait_not_in_text("#nav-system", "Dashboard")
```
Comment (martinpitt) [inline test/verify/check-host-switching:265]: This will most probably fail on -distropkg.

**Review by martinpitt — DISMISSED** [review]
Looks good, thanks! Likely needs a -distropkg exception, but I'm happy to re-review quickly.

File: test/verify/check-host-switching (hunk 1 above)
Comment (mvollmer) [reply test/verify/check-host-switching]: Yes, fixed.

**Review by martinpitt — APPROVED** [review]
Thanks!

File: test/verify/check-host-switching [hunk 2]
```diff
@@ -258,6 +261,10 @@ class TestHostSwitching(MachineCase, HostSwitcherHelpers):
         b.click("#nav-hosts .nav-item a[href='/@10.111.113.3']")
         b.wait_js_cond('window.location.pathname.indexOf("/@10.111.113.3") === 0')
 
+        # remote hosts should not have the dashboard, since #14590
+        if m.image != "rhel-8-3-distropkg": # Introduced in #14567
+            b.wait_not_in_text("#nav-system", "Dashboard")
```
Comment (martinpitt) [inline test/verify/check-host-switching:266]: This special case isn't actually necessary any more, but after this lands I'll rebase #14608 and clean it up.

**Pull request merged**